### PR TITLE
fix: [ANDROAPP-3367] Update granular sync default error message and o…

### PR DESCRIPTION
**Pull request by Balcan**
…pen logs

## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3367)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/utils/granularsync/SyncStatusDialog.kt [hunk 1]
```diff
@@ -532,7 +539,7 @@ class SyncStatusDialog : BottomSheetDialogFragment(), GranularSyncContracts.View
             }
             WorkInfo.State.FAILED -> {
                 val listStatusLog = ArrayList<StatusLogItem>()
-                if (workInfo.outputData.keyValueMap["conflict"] != null) {
+                if (/*workInfo.outputData.keyValueMap["conflict"] != null*/false) {
```
Comment (mmmateos) [inline app/src/main/java/org/dhis2/utils/granularsync/SyncStatusDialog.kt:542]: Commented code

File: app/src/main/java/org/dhis2/utils/granularsync/GranularSyncPresenterImpl.kt [hunk 2]
```diff
@@ -481,4 +484,28 @@ class GranularSyncPresenterImpl(
 
     override fun displayMessage(message: String?) {
     }
+
+    override fun syncErrors(): List<ErrorViewModel> {
+        val errors: MutableList<ErrorViewModel> = arrayListOf()
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/utils/granularsync/GranularSyncPresenterImpl.kt:489]: why mutable being just local to the function?

**Review by mmmateos — APPROVED** [review]

File: app/src/main/java/org/dhis2/utils/granularsync/GranularSyncPresenterImpl.kt [hunk 3]
```diff
@@ -481,4 +484,27 @@ class GranularSyncPresenterImpl(
...
+            addAll(
+                errorMapper.mapFKViolation(
+                    d2.maintenanceModule().foreignKeyViolations().blockingGet()
+                )
+            )
+            sortByDescending {
+                it.creationDate?.time
+            }
+        }
+    }
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/utils/granularsync/GranularSyncPresenterImpl.kt:509]: I'm approving it ! but I would have put it simpler these lines, though

val list1 = Z
val list2 = Y                
val list3 = X

return whateverTypeListOf(X,Y,Z). sortByDescending {it.creationDate?.time}

**Review by JaimeToca — APPROVED** [review]

File: app/src/main/java/org/dhis2/utils/granularsync/GranularSyncPresenterImpl.kt (hunk 3 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/utils/granularsync/GranularSyncPresenterImpl.kt:509]: that won't work. arrayListOf does not accept lists as arguments and sortByDescending returns a Unit.

**Pull request merged**